### PR TITLE
[SofaSimpleFem][SofaGeneralSimpleFem][SofaMiscFem] Avoid vectors normalization

### PR DESCRIPTION
**Pull request by alxbilger**
Some vector calculus are performed on normalized orthogonal vectors, and the result is also normalized. Therefore, no need to normalize it again. The proposed change avoids 2 normalizations. To be multiplied by the number of elements. It's called in `addForce`. And I suspect this is why `addForce` is faster in https://github.com/sofa-framework/sofa/pull/2550 (the branch contains the same changes only for TetrahedronFEMForceField).
In this PR, the changes are applied on all the corotational FEM I could found.

I found that the function `Decompose<Real>::getRotation` is very similar to my changes. However, the resulting rotation is the transpose of the rotation needed in the FEM. Also, I did not find any use of `Decompose<Real>::getRotation`.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by alxbilger** [comment]
I benchmarked only TetrahedronFEMForceFieldScene:

Before
```
------------------------------------------------------------------------------------------------------------------------
Benchmark                                                              Time             CPU   Iterations UserCounters...
------------------------------------------------------------------------------------------------------------------------
BM_Scene_bench_StepFactor<TetrahedronFEMForceFieldScene>/512        3197 ms         3188 ms            1 FPS=160.627/s frame=6.22559ms
BM_Scene_bench_StepFactor<TetrahedronFEMForceFieldScene>/1024       6434 ms         6438 ms            1 FPS=159.068/s frame=6.28662ms
BM_Scene_bench_StepFactor<TetrahedronFEMForceFieldScene>/2048      12923 ms        12938 ms            1 FPS=158.3/s frame=6.31714ms
BM_Scene_bench_StepFactor<TetrahedronFEMForceFieldScene>/4096      26074 ms        26062 ms            1 FPS=157.161/s frame=6.36292ms
```

After

```
------------------------------------------------------------------------------------------------------------------------
Benchmark                                                              Time             CPU   Iterations UserCounters...
------------------------------------------------------------------------------------------------------------------------
BM_Scene_bench_StepFactor<TetrahedronFEMForceFieldScene>/512        3128 ms         3125 ms            1 FPS=163.84/s frame=6.10352ms
BM_Scene_bench_StepFactor<TetrahedronFEMForceFieldScene>/1024       6343 ms         6344 ms            1 FPS=161.419/s frame=6.19507ms
BM_Scene_bench_StepFactor<TetrahedronFEMForceFieldScene>/2048      12693 ms        12656 ms            1 FPS=161.817/s frame=6.17981ms
BM_Scene_bench_StepFactor<TetrahedronFEMForceFieldScene>/4096      25060 ms        25062 ms            1 FPS=163.431/s frame=6.11877ms
```

It's not very significative since it is run only once, but it tends to confirm that the changes do accelerate the simulations.

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Comment by alxbilger** [comment]
One other condition on the fact it is not necessary to normalize again after a cross product is that both input vectors must be orthogonal. This has to be checked before accepting this PR.

File: SofaKernel/modules/SofaSimpleFem/src/SofaSimpleFem/HexahedronFEMForceField.inl [hunk 1]
```diff
@@ -833,11 +833,9 @@ void HexahedronFEMForceField<DataTypes>::computeRotationLarge( Transformation &r
     edgex.normalize();
     edgey.normalize();
 
-    Coord edgez = cross( edgex, edgey );
-    edgez.normalize();
+    const Coord edgez = cross( edgex, edgey );
```
Comment (epernod) [inline SofaKernel/modules/SofaSimpleFem/src/SofaSimpleFem/HexahedronFEMForceField.inl]: edgez = edgex ∧ edgey  = ‖ edgex  ‖ × ‖ edgey   ‖ × sin ⁡ ( θ ) × c   (c is unitary in edgez  direction) 
so edgez might not be always normalized if the hexahedron is deformed. However as the tests are all green this need further investigations!

**Review by epernod — CHANGES_REQUESTED** [review]

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

File: SofaKernel/modules/SofaSimpleFem/src/SofaSimpleFem/HexahedronFEMForceField.inl (hunk 1 above)
Comment (alxbilger) [reply SofaKernel/modules/SofaSimpleFem/src/SofaSimpleFem/HexahedronFEMForceField.inl]: I considered your remark. Let me know what you think

**Review by epernod — APPROVED** [review]
Ok for me, let have a double check from someone else.

**Pull request merged**